internal/storage/bucket_handle_test.go

```diff
@@ -366,6 +367,89 @@ func (t *BucketHandleTest) TestCreateObjectMethodWithGenerationAsZeroWhenObjectA
 	AssertTrue(errors.As(err, &precondition))
 }
 
+func (t *BucketHandleTest) TestCreateChunkUploaderMethodWithValidObject() {
+	chunkSize := googleapi.MinUploadChunkSize
+	req := &gcs.CreateObjectRequest{
+		Name: "test_object",
+	}
+	content := "Creating a new object"
+	contents := strings.NewReader(content)
+
+	uploader, err := t.bucketHandle.CreateChunkUploader(context.Background(),
+		req,
+		chunkSize,
+		nil,
+	)
+
+	AssertEq(nil, err)
+	AssertNe(nil, uploader)
+	AssertEq(0, uploader.BytesUploadedSoFar())
+
+	err = uploader.Upload(context.Background(), contents)
+
+	AssertEq(nil, err)
+
+	o, err := uploader.Close(context.Background())
+
+	AssertEq(nil, err)
+	AssertNe(nil, o)
+	AssertEq("test_object", o.Name)
+	AssertEq(len(content), o.Size)
+}
+
+func (t *BucketHandleTest) TestCreateChunkUploaderMethodWithGenerationAsZero() {
+	var generation int64 = 0
+	chunkSize := googleapi.MinUploadChunkSize
+	req := &gcs.CreateObjectRequest{
+		Name:                   "test_object",
+		GenerationPrecondition: &generation,
+	}
+	ctx := context.Background()
+	content := "Creating a new object"
+	contents := strings.NewReader(content)
+
+	uploader, err := t.bucketHandle.CreateChunkUploader(ctx,
+		req,
+		chunkSize,
+		nil)
+
+	AssertEq(nil, err)
+	AssertNe(nil, uploader)
+	AssertEq(0, uploader.BytesUploadedSoFar())
+
+	err = uploader.Upload(context.Background(), contents)
+
+	AssertEq(nil, err)
+
+	o, err := uploader.Close(context.Background())
+
+	AssertEq(nil, err)
+	AssertNe(nil, o)
+	AssertEq("test_object", o.Name)
+	AssertEq(len(content), o.Size)
+}
+
+func (t *BucketHandleTest) TestCreateChunkUploaderMethodWithGenerationAsNonZero() {
+	var generation int64 = 47367842389354
+	chunkSize := googleapi.MinUploadChunkSize
+	expectedErrorSubstring := "method not supported for pre-existing GCS objects"
+	req := &gcs.CreateObjectRequest{
+		Name:                   "test_object",
+		GenerationPrecondition: &generation,
+	}
+
+	// this should fail as CreateChunkUploader is currently supported
+	// only to objects that don't exist already.
+	uploader, err := t.bucketHandle.CreateChunkUploader(context.Background(),
+		req, chunkSize, nil)
+
+	AssertEq(nil, uploader)
+	AssertNe(nil, err)
+	if !strings.Contains(err.Error(), expectedErrorSubstring) {
```

Comment:
nit: AssertTrue(strings.Contains(err.Error(), expectedErrorSubstring))

Reply:
This is on purpose. This is to make it easier to identify the issue directly from the error log, by printing out the expected error log and the actual error log, rather than having to debug or change the code to find out the issue. I'll keep it this way.